src/components/Geogebra/StudentClient.js

```diff
@@ -0,0 +1,453 @@
+    // the applet. Caption style means that objects are
+    // named by their captions.
+    CAPTION_STYLE: 3,
+
+    POINT:         'point',
+    TEXTFIELD:     'textfield',
+    NUMERIC:       'numeric',
+
+    /**
+     * GGB commands have the following
+     * format: <object-label>:<command-body>.
+     *
+     * @param {String} label
+     * @param {String} cmdStrBody
+     */
+    getCommand(label, cmdStrBody) {
+        return `${label}:${cmdStrBody}`;
+    }
+};
+
+/**
+ * Throttle ensures that function [func] is evaluated at most once per
+ * [limit] period.
+ */
+function throttle (func, limit) {
+    let inThrottle;
+
+    return function() {
+        const args = arguments;
+        const context = this;
+
+        if (!inThrottle) {
+            func.apply(context, args);
+            inThrottle = true;
+            setTimeout(() => inThrottle = false, limit);
+        }
+    }
+}
+
+/**
+ * Debounce ensures that the function [func] is evaluated after [delay]
+ * time since the last call to the function (classic debounce behavior).
+ */
+function debounce (func, delay) {
+    // [inDebounce] is a variable used to track the delay period.  If
+    // invoked for the first time, the [func] will execute at the end of
+    // the delay.  If invoked and then invoked again before the end of the
+    // delay, the delay restarts.
+    let inDebounce;
+
+    return function() {
+        const context = this;
+        const args = arguments;
+
+        clearTimeout(inDebounce);
+        inDebounce = setTimeout(() => func.apply(context, args), delay);
+    }
+}
+
+class StudentClient {
+    /**
+     * @param {Object} params Geogebra applet parameters
+     * (https://wiki.geogebra.org/en/Reference:GeoGebra_App_Parameters).
+     *
+     * @param {Object} log Vue logger.
+     */
+    constructor(params, log) {
+        this.params = {
+            container: 'geogebra_designer',
+            id: 'applet',
+            width: 800,
+            height: 600,
+            perspective: 'AG',
+            showAlgebraInput: true,
+            showToolBarHelp: false,
+            showMenubar: true,
+            enableLabelDrags: false,
+            showResetIcon: true,
+            showToolbar: true,
+            allowStyleBar: false,
+            useBrowserForJS: true,
+            enableShiftDragZoom: true,
+            errorDialogsActive: true,
+            enableRightClick: false,
+            enableCAS: false,
+            enable3d: false,
+            isPreloader: false,
+            screenshotGenerator: false,
+            preventFocus: false,
+            scaleContainerClass: 'appletContainer',
+            resizeFactor: 1.0,
+            // Initial Geogebra state encoded in Base64
+            initialState,
+            ...params,
+        };
+
+        this.log = this.params.log;
+        this.params.listener = undefined;
+        this.appletContainer = new GGBApplet(this.params);
+        this.ignoreUpdates = false;
```

Comment:
I would appreciate a short comment on what's that and how it's used. This flag creeps into many methods down there so it would be good to have a solid understanding of its behaviour.